File: Python/p3.py
r, c = map(int, input("Enter rows and columns: ").split())
z = 0
print("Enter elements:")
for i in range(r):
    row = list(map(int, input().split()))
    z += row.count(0)

if z > (r * c) // 2:
    print("Matrix is Sparse")
else:
    print("Matrix is Not Sparse")

File: C/p3.c
#include <stdio.h>

int main() {
    int r, c, z = 0;
    printf("Enter rows and columns: ");
    scanf("%d %d", &r, &c);

    int a[r][c];

    printf("Enter elements:\n");
    for(int i = 0; i < r; i++) {
        for(int j = 0; j < c; j++) {
            scanf("%d", &a[i][j]);
            if(a[i][j] == 0) {
                z++;
            }
        }
    }

    if(z > (r * c) / 2) {
        printf("Matrix is Sparse\n");
    } else {
        printf("Matrix is Not Sparse\n");
    }

    return 0;
}
